src/main/features/mediation/routes/mediation-agreement.ts

```diff
@@ -0,0 +1,52 @@
+import * as express from 'express'
+
+import { Paths } from 'mediation/paths'
+import { Paths as ResponsePaths } from 'response/paths'
+import { Paths as ClaimantResponsePaths } from 'claimant-response/paths'
+import { ErrorHandling } from 'main/common/errorHandling'
+import { Claim } from 'claims/models/claim'
+import { Draft } from '@hmcts/draft-store-client'
+import { MediationDraft } from 'mediation/draft/mediationDraft'
+import { FreeMediation, FreeMediationOption } from 'main/app/forms/models/freeMediation'
+import { DraftService } from 'services/draftService'
+
+function renderView (res: express.Response): void {
+  const user: User = res.locals.user
+  const claim: Claim = res.locals.claim
+
+  res.render(Paths.mediationAgreementPage.associatedView, {
+    otherParty: claim.otherPartyName(user)
+  })
+
+}
+
+/* tslint:disable:no-default-export */
+export default express.Router()
+  .get(Paths.mediationAgreementPage.uri, (req: express.Request, res: express.Response) => {
+    renderView(res)
+  })
+  .post(
+    Paths.mediationAgreementPage.uri,
+    ErrorHandling.apply(async (req: express.Request, res: express.Response) => {
+      const claim: Claim = res.locals.claim
+      const draft: Draft<MediationDraft> = res.locals.mediationDraft
+      const user: User = res.locals.user
+
+      if (req.body.reject) {
```

Comment:
The current logic may assume it is accepted if got post from any other place (other than `mediation-agreement.njk`.
How about deciding if we have `req.body.accept` rather than `reject`? It will be more safe.